### PR TITLE
[fix]: [CDS-54147]: adding rest api resource for helm manifest to support ex…

### PR DESCRIPTION
**Pull request by markoharness**
…pression resolution for connector and chart name when chart version is runtime input

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44751)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java [hunk 1]
```diff
@@ -63,4 +70,44 @@ public ResponseDTO<HelmChartResponseDTO> getHelmChartVersionDetails(
         projectIdentifier, serviceRef, fqnPath, connectorIdentifier, chartName, region, bucketName, folderPath);
     return ResponseDTO.newResponse(helmChartResponseDTO);
   }
+
+  @GET
```
Comment (abosii-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java:74]: Shouldn't this be POST?

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java (hunk 1 above)
Comment (markoharness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java:74]: I would say no. 
This is just a querying method and does not makes any changes(inserts, updates nor deletes).
Is there any reason that it should be a POST?

**Comment by markoharness** [comment]
trigger messagecheck

File: 120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java (hunk 1 above)
Comment (abosii-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java:74]: I don't really remember all details, but I believe some parts of HTTP specification forbid or not recommend using GET with message payload. It's mostly related to the caching mechanism, as a GET method is defined by spec as identifiable through request url (including query params) and payload is not part of this. Some implementations may reject GET with payload so is safer to just use POST in this case as your response is based on the content of the request

**Comment by markoharness** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java` line 74 at r1](https://reviewable.io/reviews/harness/harness-core/44751#-NPqnVrkbVzgEK4SoT36-r1-74:-NPrRG826q_pv79_E6Kp:b-oxqj78) ([raw file](https://github.com/harness/harness-core/blob/49d92beaab0af21da7d5b92399c2611d1faf656c/120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java#L74)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

I don't really remember all details, but I believe some parts of HTTP specification forbid or not recommend using GET with message payload. It's mostly related to the caching mechanism, as a GET method is defined by spec as identifiable through request url (including query params) and payload is not part of this. Some implementations may reject GET with payload so is safer to just use POST in this case as your response is based on the content of the request
</blockquote></details>

Fair enough. I will change it to POST.

**Review by achyuth-harness — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44751)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44751)* status: 1 change requests. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java` line 87 at r2](https://reviewable.io/reviews/harness/harness-core/44751#-NPvRUe12wTW-p-5elS8:-NPvRUe12wTW-p-5elS9:b-1ffczn) ([raw file](https://github.com/harness/harness-core/blob/2bfb9bba71fb9e5e4ec44dc7f938a93e1dff57a1/120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java#L87)):*
> ```Java
>       @NotNull @QueryParam("fqnPath") String fqnPath, @QueryParam("connectorRef") String connectorIdentifier,
>       @QueryParam("chartName") String chartName, @QueryParam("region") String region,
>       @QueryParam("bucketName") String bucketName, @QueryParam("folderPath") String folderPath,
> ```

We may need to support expression for these fields as well

**Comment by markoharness** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java` line 87 at r2](https://reviewable.io/reviews/harness/harness-core/44751#-NPvRUe12wTW-p-5elS8:-NPw_AhA9aHHxJn2QI3i:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2bfb9bba71fb9e5e4ec44dc7f938a93e1dff57a1/120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java#L87)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

We may need to support expression for these fields as well
</blockquote></details>

Done.

**Review by abosii-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44751)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by namantalaycha10 — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44751)* status: :shipit: complete! ✓

**Pull request merged**